lemmy-almalinux.yml

```diff
@@ -44,6 +44,11 @@
       ansible.builtin.systemd:
         name: nginx
         state: reloaded
+
+    - name: Reload firewalld
```

Comment:
Hmm, so basically with `immediate: true` we don't need to an explicit reload at all? The module will do it on its own?

Reply:
Yes, that is correct. 

It might help to understand what `permanent` means in the context of `firewallD`. 

This command will allow `http` traffic in the `public` zone and take effect *immediately*. The rule will *not* be persistent across reboots/reloads.

```shell
firewall-cmd --zone=public --add-service=http
````
Comparatively, the changes are saved but not applied when you use the `--permanent` flag (which is what the modules `permanent: true` refers to). As such, you'd usually pair it with `firewall-cmd --reload`:

```shell
firewall-cmd --zone=public --add-service=http --permanent # Not applied yet
firewall-cmd --reload # Now it's applied
```
So, in short, when you pair `immediate: true` with `permanent: true`, it effectively does the above, making the reload redundant. 

There is a caveat with this when handling zones, requiring you to reload firewalld explicitly if you want to use it immediately [1]. However, it's unnecessary since we use the default `public` zone.

[1] https://docs.ansible.com/ansible/latest/collections/ansible/posix/firewalld_module.html#notes

Reply:
Hmm okay, I'll keep the handler for now then but will remove the notify